File: IamStupendous.Resume/src/IamStupendous.Resume/wwwroot/scripts/app/work/work.ts
import { Project } from './project'

/**
 * Work model
 */
export class Work {
    company: string;
    projects: Project[];
    yearStart: string;
    yearEnd: string;
    isOpen: boolean;

    constructor(company: string,
                yearStart: string,
                yearEnd: string,
                projects: Project[]) {
        this.company = company;
        this.yearStart = yearStart;
        this.yearEnd = yearEnd;
        this.projects = projects;
        this.isOpen = false;
    }
}

File: IamStupendous.Resume/src/IamStupendous.Resume/Models/Work.cs
using System;
using System.Collections.Generic;

namespace IamStupendous.Resume.Models
{
    public class Work
    {
        public string Company { get; set; }
        public IList<Project> Projects { get; set; }
        public DateTime YearStart { get; set; }
        public DateTime? YearEnd { get; set; }

        public Work(string company,
            DateTime yearStart,
            DateTime? yearEnd,
            IList<Project> projects)
        {
            Company = company;
            Projects = projects;
            YearStart = yearStart;
            YearEnd = yearEnd;
        }
    }
}